### PR TITLE
Database version Rename

### PR DESCRIPTION
**Pull request by cblgh**
i renamed protocol version to database version and used an int to represent it instead as it doesn't make sense to have minor & patch on something like this.

lmk what else we can do here before we merge into master

this will be required for [cabal-cli's multi cabal pr](https://github.com/cabal-club/cabal/pull/89)

### CODE REVIEW

**Comment by karissa** [comment]
What other modules break by changing the external api for `protocolVersion` to `databaseVersion`? This would probably be a major bump of cabal-core.

Using a separate version number than the core library that's published on npm might cause confusion later in time. With a small distributed set of contributors, reducing the number of variables at play can help with groking potential bugs or mismatches in versions. What if we used the version from package.json?

**Comment by cblgh** [comment]
@karissa nothing will break as nobody is relying on protocolVersion yet, it's just the multi-cabal pr right now. which is why i felt okay with doing a breaking rename like this

see the discussion here https://github.com/cabal-club/cabal-core/pull/25

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "cabal-core",
-  "version": "3.0.4",
+  "version": "3.1.0",
```
Comment (noffle) [inline package.json]: Could we do version bumps separately from PRs?

File: index.js [hunk 2]
```diff
@@ -66,9 +66,9 @@ function Cabal (storage, key, opts) {
 }
 
 inherits(Cabal, events.EventEmitter)
-Cabal.prototype.getProtocolVersion = function (cb) {
+Cabal.prototype.getDatabaseVersion = function (cb) {
```
Comment (noffle) [inline index.js]: Is the code that changes the `~/.cabal/archives` dir name already in master, or can that get into this PR too?

File: package.json (hunk 1 above)
Comment (cblgh) [reply package.json]: lol yeah sorry, i keep thinking it's a good place to do in PRs. fixing!

File: index.js (hunk 2 above)
Comment (cblgh) [reply index.js]: @noffle it's not in master, but in the multi-cabal pr here https://github.com/cabal-club/cabal/pull/89

(i'm not sure how i would bring it in the cabal-core pr)

Comment (noffle) [reply index.js]: Ah, of course!

My thinking was "gosh it'd be nice if we could get in the db-dir changes so we can test that separately from the multi-cabal work".

**Pull request merged**